azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/handlers/FunctionCoreToolsHandlerImpl.java

```diff
@@ -60,38 +53,47 @@ protected void installFunctionExtension() throws Exception {
         );
     }
 
-    protected boolean isLocalVersionSupportAutoInstall(final String localVersion) {
+    protected boolean checkLocalCoreToolsVersion() {
```

Comment:
Let's change the function to `void assureRequirementAddressed()`

And do three steps in it:
- check function cli installed
- check function cli has the ability to install extensions
- check function cli is not out-dated